diesel_derives/src/queryable_by_name.rs

```diff
@@ -38,8 +38,36 @@ pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagno
 
     let (impl_generics, _, where_clause) = generics.split_for_impl();
 
+    let impl_table_queryable: Option<syn::Item> = if model.has_table_name_attribute() {
```

Comment:
The more I think about this, the more I'm the opinion that this should be a separate derive for the following reasons:

* We do also need a `derive(Queryable)` on the same type otherwise (by having a separate derive, we can just also implement `Queryable` there)
* How would we handle the case of deriving `QueryableByName` where only `#[sql_type = "Foo"]` is given? I think just ignoring everything without table name attribute creates a wired edge case.

Reply:
Fair enough, we could also benifit from separate `derive` for join scenario.